pennylane/devices/qubit/simulate.py

```diff
@@ -296,6 +323,12 @@ def simulate(
 
     has_mcm = any(isinstance(op, MidMeasureMP) for op in circuit.operations)
     if circuit.shots and has_mcm:
+        if execution_kwargs.get("mcm_method", None) == "tree-traversal":
+            n_mcms = sum(isinstance(op, MidMeasureMP) for op in circuit.operations)
+            if 2 * n_mcms + 100 > sys.getrecursionlimit():
+                sys.setrecursionlimit(2 * n_mcms + 100)
```

Comment:
Do we need to reset this after the simulation is complete?

Reply:
I can be safer for users, let me do it.